tests/decorator-free.test.ts

```diff
@@ -0,0 +1,321 @@
+/* eslint-disable @typescript-eslint/unbound-method */
+import { ConfiguredInstance, DBOS } from '../src/';
+import { generateDBOSTestConfig, setUpDBOSTestDb } from './helpers';
+import { randomUUID } from 'node:crypto';
+
+function stepTest(value: number): Promise<number> {
+  expect(DBOS.stepStatus).toBeDefined();
+  return Promise.resolve(value * 100);
+}
+
+const retryTestAttempts = Array<boolean>(5).fill(false);
+function retryTest(value: number): Promise<number> {
+  expect(DBOS.stepStatus!.currentAttempt).toBeDefined();
+  retryTestAttempts[DBOS.stepStatus!.currentAttempt!] = true;
+  if (DBOS.stepStatus!.currentAttempt! < 3) {
+    throw new Error('retry-test-error');
+  }
+  return Promise.resolve(value * 100);
+}
+
+const regStepTest = DBOS.registerStep(stepTest);
+const regRetryTest = DBOS.registerStep(retryTest, { retriesAllowed: true });
+
+function wfRegStep(value: number) {
+  return regStepTest(value);
+}
+
+function wfRunStep(value: number) {
+  return DBOS.runStep(() => stepTest(value), { name: 'stepTest-runStep' });
+}
+
+function wfRegRetry(value: number) {
+  return regRetryTest(value);
+}
+
+const regWFRegStep = DBOS.registerWorkflow(wfRegStep, 'wfRegStep');
```

Comment:
Is there a check somewhere if someone tries to reuse a workflow name?

Reply:
As I was copy pasting test code, I forgot to change the workflow name in one case and the test crashed on startup. I'll look to see what it would take to make a unit test for that. Also want to have a test using queue to ensure that we can correctly dispatch to registered workflows by string

Reply:
I added a test for this + a single enqueue for free functions. Will add static/instance method enqueue tests before merging